pkg/es/pongo.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package es
+
+import "github.com/flosch/pongo2/v4"
```

Comment:
Why do we need this new template engine? Do we need _any_ template engine for the functionality that's being introduced? Why can't we use stdlib text/template?

Reply:
@yurishkuro, the motivation is to avoid introducing another index mapping file to enable ILM functionality (we currently already maintain one per version (6 or 7), per index-type (span or service) = 4 files). Prior to this, the index mapping file was templated with some simple variables, which just involved a simple search + replace operation. 

With the need for slightly more complex conditional logic, using a templating engine avoids the need for re-inventing the wheel in code. Do you see any potential problems using template engines?

Thanks for the `text/template` suggestion; I think it would be a better alternative to bringing in a new dependency. I see Python has an equivalent port of the go templating engine as well: https://pypi.org/project/go-template/.

Reply:
We need to always weigh the pros and cons of introducing new dependencies, because they both bloat the binary size and just create an additional mental complexity. From my cursory overview of the usage, I did not see why the standard text/template would not work.

Is there some requirement for the template files to work both for Go and Python? Can we not converge into one language, e.g. build a small util in Go to be used from Python?

Reply:
> We need to always weigh the pros and cons of introducing new dependencies

Okay, I'll keep this in mind in future, thanks.

> From my cursory overview of the usage, I did not see why the standard text/template would not work.

I agree that `text/template` would work; we basically need support for `if ... then ... end`.

> Is there some requirement for the template files to work both for Go and Python? Can we not converge into one language, e.g. build a small util in Go to be used from Python?

IIUC, the only requirement is for the python script to get the resulting "filled out" json blob as a string; how this is done does not matter so yes, we can converge into one language and write a small Go util that transforms the template to JSON.

Reply:
@albertteoh Should we actually have the rollover script itself in golang. Not sure why we went with python in first place.

Reply:
This was also before my time with Jaeger, but my guess is because of its dependency on [curator](https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/esRollover.py#L4) which seems to be a python library: https://github.com/elastic/curator. Maybe @pavolloffay or @yurishkuro know?

Reply:
Sorry, Could not find time to work on this feedback. Will start this week.

To summarise the changes to be done - 
- we go with golang text/template.
- Build a go binary to render the template to json.
- Copy binary to python image and use to get the jsons.
- Use go binary to render the template jsons in python.
- Cleanup pongo dependency and fix corresponding tests.

@albertteoh @yurishkuro  - Please let me know, if I missed anything.

Reply:
@albertteoh  - I have implemented the feedback on templates. Have written a small golang util which would render the templates and print it to stdout. This util is then used in python to get the rendered output. I felt this is a simple way of doing it. 

We could look at creating a shared library (.so) from golang (using Cgo) and using it from python [rather than printing it out to stdout] - let me know if you feel we need to go that way.

Reply:
@bhiravabhatla, sorry for the late reply on this. I like your idea of creating a shared lib, as I don't think running this as a standalone executable makes much sense; I can't think of any use cases for it right now. It also does feel a little more clunky to run the tool in a shell and read stdout, instead of calling it natively.

The only consideration is, at least from my local testing, the `.so` file is about 15% larger than the executable version from 13M to 15M. I'm not sure if this is considered significant.

Reply:
@albertteoh  -  I ll try out shared library way, I agree that current approach is little hacky :). I am new to the whole shared library thing. Will reach out if I am stuck. Thanks!

Reply:
Personally I don't mind just calling into a child process, I think it's much easier to debug if something goes wrong than to debug a shared library written in Go and called from Python (not sure it's even possible to call).

However, if the util is printing to stdout, it may be more complicated to handle from Python than just passing it a file name to write. But not _that much_ more complicated.

Reply:
@yurishkuro - I was able to capture and use the string printed out to stdout in python without any issues. I am happy to stay with current approach. @albertteoh & @yurishkuro  your call :)

Reply:
@bhiravabhatla Okay, in that case, please stick with the current implementation of printing to stdout, especially since it sounds like we're all pretty new to this approach. :) We can revisit later if necessary.